keps/sig-node/3317-user-namespaces/README.md

```diff
@@ -0,0 +1,1015 @@
+- `pkg/volume/downwardapi/downwardapi.go`: 24.05.2022 - 61.7%
+- `pkg/volume/secret/secret.go`: 24.05.2022 - 65.7%
+- `pkg/volume/projected/projected.go`: 24.05.2022 - 68.2%
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+We can do e2e test to verify that userns is disabled when it should.
+
+To test with userns enabled, we need to patch container runtimes. We can either
+try to use a patched version or make the alpha longer and add e2e when we can
+use container runtime versions that have the needed changes.
+
+##### critests
+
+- For Alpha, the feature is tested for containerd and CRI-O in cri-tools repo using critest to
+  make sure the specified user namespace configuration is honored.
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+Note this section is a WIP yet.
+
+##### Alpha
+- Phase 1 implemented
+
+##### Beta
+
+- Make plans on whether, when, and how to enable by default
+- Should we reconsider making the mappings smaller by default?
+- Should we allow any way for users to for "more" IDs mapped? If yes, how many more and how?
+- Should we allow the user to ask for specific mappings?
+
+##### GA
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Some definitions first:
+- New kubelet: kubelet with CRI proto files that includes the changes proposed in
+this KEP.
+
+- Old kubelet: idem, but CRI proto files don't include this changes.
+
+- New runtime: container runtime with CRI proto files that includes the changes
+proposed in this KEP.
+
+- Old runtime: idem, but CRI proto files don't include this changes.
+
+New runtime and old kubelet: all works just fine. Kubelet doesn't request userns
```

Comment:
"all works just fine" is misleading.

What you really say is effectively "the feature is disabled", right?

Reply:
Yes, I will clarify. Thanks

Reply:
Chaing "all works just fine" to "containers are created without user namespaces"